src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -415,7 +415,7 @@ public virtual void Write(TextWriter writer, TextSpan span, CancellationToken ca
             }
         }
 
-        internal ImmutableArray<byte> GetChecksum(bool useDefaultEncodingIfNull = false)
+        public ImmutableArray<byte> GetChecksum(bool useDefaultEncodingIfNull = false)
```

Comment:
Except, the one that is exposed should not lead to an exception being thrown if no encoding is known.   So the public method should always have useDefaultEncodingIfNull as true.

---
In reply to: [72319541](https://github.com/dotnet/roslyn/pull/12757#discussion_r72319541) [](ancestors = 72319541)